docs/errors/E0717.md

```diff
@@ -0,0 +1,46 @@
+# E0717: using a '.' after a '?.' might fail, since '?.' might return 'undefined'
+
+```config-for-examples
+{
+  "globals": {
+  }
+}
+```
+
+In Javascript, the `?.` operator performs a conditional access only if the left-hand side is not `null` or `undefined`. If the left-hand side is `null` or `undefined`, the expression returns `undefined` and the right-hand side is not evaluated. Using it suppresses the error that would otherwise be thrown if the left-hand side were `null` or `undefined` and the operator `.` were used instead. However, if the left-hand side is not `null` or `undefined`, the expression returns the value of the right-hand side, which may be `null` or `undefined`. If the right-hand side is `null` or `undefined`, using the `.` operator will throw an error:
+
+```js
+let a = { b: 1 };
+let b = a?.b; // ok
+let c = a?.b?.a; // ok
+let d = a?.b?.a.c; // throws an error
```

Comment:
Nit: This example is confusing to me at a glance because there are multiple things named `a` and `b`. I think this example would be a bit clearer with `console.log` and different names:
```suggestion
let a = { b: 1 };
console.log(a?.b);      // 1
console.log(a?.b?.c);   // undefined
console.log(a?.b?.c.d); // throws an error
```
Even better would be to use more realistic names you might see in a program:
```suggestion
let bug = { milestone: null  };
console.log(bug.milestone);               // null
console.log(bug.milestone?.name);         // undefined
console.log(bug.milestone?.name.trim());  // throws an error
```